G-Research.OpenTelemetry.Processor.Partial/SpecHelper.cs

```diff
@@ -0,0 +1,31 @@
+using System.Text.Encodings.Web;
+using System.Text.Json;
+
+namespace GR.OpenTelemetry.Processor.Partial;
+
+public class SpecHelper
+{
+    public static string Json(TracesData tracesData) =>
+        JsonSerializer.Serialize(tracesData, new JsonSerializerOptions
+        {
+            PropertyNamingPolicy = JsonNamingPolicy.SnakeCaseLower,
+            WriteIndented = false, // For pretty printing
+            Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping, // To allow special characters
+            DefaultIgnoreCondition =
+                System.Text.Json.Serialization.JsonIgnoreCondition
+                    .WhenWritingNull
+        });
+
+    public static ulong ToUnixTimeNanoseconds(DateTime dateTime)
+    {
+        dateTime = dateTime.ToUniversalTime();
+
+        DateTime epoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+
+        // Ticks since epoch (1 tick = 100ns)
+        long ticksSinceEpoch = dateTime.Ticks - epoch.Ticks;
+
+        // Convert to nanoseconds (multiply by 100)
+        return (ulong)(ticksSinceEpoch * 100);
+    }
```

Comment:
Inspired by:
[`DateTimeOffset.ToUnixTimeMilliseconds()`](https://github.com/dotnet/runtime/blob/1d1bf92fcf43aa6981804dc53c5174445069c9e4/src/libraries/System.Private.CoreLib/src/System/DateTimeOffset.cs#L825C13-L826C57)
[`DateTime.UnixEpochTicks`](https://github.com/dotnet/runtime/blob/0b6404153e16685f47d166625373f6635cf5631d/src/libraries/System.Private.CoreLib/src/System/DateTime.cs#L86) (which is `internal`! 😠)
[`CborHelpers.UnixEpochTicks`](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Formats.Cbor/src/System/Formats/Cbor/CborHelpers.netstandard.cs#L17) (which is private but easier to copy)

```suggestion
    private const long UnixEpochTicks = 719162L /*Number of days from 1/1/0001 to 12/31/1969*/ * 10000 * 1000 * 60 * 60 * 24; /* Ticks per day.*/
    
    public static ulong ToUnixTimeNanoseconds(DateTime dateTime)
    {
        dateTime = dateTime.ToUniversalTime();

        long ticksSinceEpoch = dateTime.Ticks - UnixEpochTicks ;

        // Convert to nanoseconds
        return (ulong)(ticksSinceEpoch * TimeSpan.NanosecondsPerTick );
    }
```

Reply:
after commiting this change, i'm getting an error for net6.0
```
Error: /Users/runner/work/otel-partial-dotnet/otel-partial-dotnet/G-Research.OpenTelemetry.Processor.Partial/SpecHelper.cs(28,51): error CS0117: 'TimeSpan' does not contain a definition for 'NanosecondsPerTick' [/Users/runner/work/otel-partial-dotnet/otel-partial-dotnet/G-Research.OpenTelemetry.Processor.Partial/G-Research.OpenTelemetry.Processor.Partial.csproj::TargetFramework=net6.0]
```

do you mind me reverthing this change and keeping the old version if you prefered this only for readability?